experimental/dds/tree2/src/feature-libraries/optional-field/optionalField.ts

```diff
@@ -316,132 +324,111 @@ export const optionalChangeRebaser: FieldChangeRebaser<OptionalChangeset> = {
-										over.fieldChange?.revision ?? overTagged.revision,
-										revisionMetadata,
-									),
-									localId: over.fieldChange.id,
-								},
-								rebasedChild,
-							);
-						}
-					} else {
-						// `over` didn't remove the node (its fieldChange is undefined)
-						const rebasedChild = rebaseChild(
-							childChange,
-							overChildChange,
-							NodeExistenceState.Alive,
-						);
-						if (rebasedChild !== undefined) {
-							perChildChanges.set(id, rebasedChild);
-						}
-					}
+		let reservedDetachId: RegisterId | undefined = change.reservedDetachId;
+		const changeDstToSrc = new ChildChangeMap<RegisterId>();
+		const changeSrcToDst = new ChildChangeMap<RegisterId>();
+		for (const [src, dst] of moves) {
+			changeSrcToDst.set(src, dst);
+			changeDstToSrc.set(dst, src);
+		}
+
+		for (const [src, dst, target] of moves) {
+			if (target === "cellTargeting") {
+				// TODO: Should we drop cell targeting / node targeting and just special-case 'self'? Might be simpler to understand.
+				// Holding off on making a call until AB#6298 is addressed (possibly support for rebasing transactions makes the
+				// answer to this more obvious).
+				if (
+					overSrcToDst.get(src) !== undefined &&
+					overDstToSrc.get(src) === undefined &&
+					reservedDetachId === undefined
+				) {
+					// Over removed the content occupying this cell and didn't fill it with other content.
+					// This change includes field changes,
+					reservedDetachId = dst;
 				} else {
-					if (
-						(restoredRollbackChangeId !== undefined &&
-							areEqualChangeIds(id, restoredRollbackChangeId)) ||
-						(restoredUndoChangeId !== undefined &&
-							areEqualChangeIds(id, restoredUndoChangeId))
-					) {
-						// childChange refers to changes to node being revived by `over`.
-						const overChange = over.fieldChange?.newContent?.changes;
-						const rebasedChild = rebaseChild(
-							childChange,
-							overChange,
-							NodeExistenceState.Alive,
-						);
-						if (rebasedChild !== undefined) {
-							perChildChanges.set("self", rebasedChild);
-						}
-					} else {
-						// childChange refers to changes to node removed by some past revision. Rebase over any changes that
-						// `over` has to that same revision.
-						const overChange = overChildChanges.get(id);
-						const rebasedChild = rebaseChild(
-							childChange,
-							overChange,
-							NodeExistenceState.Dead,
-						);
-						if (rebasedChild !== undefined) {
-							perChildChanges.set(id, rebasedChild);
-						}
-					}
+					// Cell-targeting.
+					rebasedMoves.push([src, dst, target]);
 				}
+			} else {
+				// Figure out where content in src ended up in `overTagged`
+				const rebasedSrc = overSrcToDst.get(src) ?? src;
+				// Note: we cannot drop changes which map a node to itself, as this loses the intention of the original edit
+				// (since the target kind is node targeting, it may not still map to a noop after further rebases)
```

Comment:
Nice call-out :-)
Funny coincidence: I added the same exact logic to sequence moves yesterday.